pkg/localstore/mode_set.go

```diff
@@ -179,65 +175,51 @@ func (db *DB) setSync(batch *leveldb.Batch, addr swarm.Address, mode storage.Mod
 		return 0, err
 	}
 
-	i, err = db.retrievalAccessIndex.Get(item)
-	switch {
-	case err == nil:
-		item.AccessTimestamp = i.AccessTimestamp
-		err = db.gcIndex.DeleteInBatch(batch, item)
-		if err != nil {
+	i1, err := db.retrievalAccessIndex.Get(item)
+	if err != nil {
+		if !errors.Is(err, leveldb.ErrNotFound) {
 			return 0, err
 		}
-		gcSizeChange--
-	case errors.Is(err, leveldb.ErrNotFound):
-		// the chunk is not accessed before
-	default:
-		return 0, err
-	}
-	item.AccessTimestamp = now()
-	err = db.retrievalAccessIndex.PutInBatch(batch, item)
-	if err != nil {
-		return 0, err
-	}
-
-	// Add in gcIndex only if this chunk is not pinned
-	ok, err := db.pinIndex.Has(item)
-	if err != nil {
-		return 0, err
-	}
-	if !ok {
-		err = db.gcIndex.PutInBatch(batch, item)
+		item.AccessTimestamp = now()
```

Comment:
TODO: here please take note of this. This should leave the access timestamp alone.

Reply:
and the following line also dropped

Reply:
....don't understand.... i thought you said setSync should update access timestamp

Reply:
let's please have a writeup or at least a truth table about what happens to which index under which circumstances. i find documenting this in the code not the best thing to do and it just pollutes the code with hypothetical reminders. let's start with defining, then make changes to the codebase and in the meanwhile try to keep it clean

Reply:
here it is quite simple, you update accesstimestamp if and only if the chunk iis requested

Reply:
truth table?